### PR TITLE
Adding discovery feedback functionality

### PR DESCRIPTION
**Pull request by tegefaulkes**
### Description

This PR addresses discovery feedback by adding events to the `Discovery` domain and RPC for streaming these events.

### Issues Fixed

* Fixes: ENG-28

### Tasks

- [x] 1. Add `Discovery` events relating to vertex processing.
- [x] 2. Add RPC command for streaming these `Discovery` events.
- [x] 3. Add tests validating these features.

### Final checklist

* [x] Domain specific tests
* [x] Full tests
* [x] Updated inline-comment documentation
* [x] Lint fixed
* [x] Squash and rebased
* [x] Sanity check the final build

### CODE REVIEW

**Comment by tegefaulkes** [comment]
The discovery vertex events is being handled by the `audit` domain now. There already exists a RPC handler for getting audit events so the discovery feedback backend stuff is covered. I've also added a RPC for listing the queued discovery tasks.

I think that covers everything required by the CLI to add the feedback commands. I'm going to move on to review and clean up now.

File: src/client/handlers/GestaltsDiscoveryQueue.ts
Comment (tegefaulkes) [inline src/client/handlers/GestaltsDiscoveryQueue.ts:1]: For the sake of it we should handle cancel-ability here.

Comment (tegefaulkes) [reply src/client/handlers/GestaltsDiscoveryQueue.ts:1]: Fixed.

**Comment by tegefaulkes** [comment]
This PR is just the backend functionality. The frontend work is being continued in https://github.com/MatrixAI/Polykey-CLI/pull/173.

**Comment by tegefaulkes** [comment]
I made a mistake, I merged into the wrong branch. I'm going to fix this with a manual merge.

**Comment by CMCDragonkai** [comment]
Merge manually with git then.

**Comment by CMCDragonkai** [comment]
Interesting - if you fix #... It does not update linear. But if you use fix ENG... It does link linear? Is there something that does both?

**Comment by tegefaulkes** [comment]
If you fix # and that's linked in linear then it will update linear when the github issue updates due to syncing. If you add `fixes ENG-` to the PR then it will update in linear directly. The issue is still open in linear due to it being linked to two PRs it github. so it's waiting for both to merge before the state change.

**Comment by CMCDragonkai** [comment]
In that case let's keep it to fixing #... - it maintains compatibility.

**Comment by tegefaulkes** [comment]
I'm using both `#` and `eng-` style. I didn't include both here since the issue isn't actually resolved until this PR and the CLI PR is done.

**Comment by CMCDragonkai** [comment]
Isn't just #... Doing double duty?

**Pull request merged**